File: nodejs/Board.js
const Square = require('./Square');

module.exports = class Board {
    constructor(n) {
        this.cols = n;
        this.squares = Board.CreateBoard(n);
    }

    AddPuzzlePiece(puzzle, index) {
        this.squares[index].puzzle = puzzle;
    }

    static CreateBoard(n) {
        const squares = [];
        for (let row = 0; row < n; row++) {
            for (let col = 0; col < n; col++) {
                squares.push(new Square(row, col));
            }
        }

        return squares;
    }
}

File: go/board.go
package main

type Board struct {
	Squares  []Square
	Diagonal int
}

func (board *Board) AddPuzzle(puzzle Puzzle, index int) {
	board.Squares[index].Puzzle = puzzle
}

func NewBoard(diagonal int) Board {
	var squares []Square

	for row := 0; row < diagonal; row++ {
		for col := 0; col < diagonal; col++ {
			squares = append(squares, Square{Row: row, Col: col})
		}
	}

	return Board{Squares: squares, Diagonal: diagonal}
}
